### PR TITLE
[VEN-228] Claim vault rewards mutations

### PR DESCRIPTION
**Pull request by therealemjy**
Add mutations to claim rewards on various vault types.

### CODE REVIEW

File: src/clients/api/mutations/useClaimVaiVaultReward.ts [hunk 1]
```diff
@@ -0,0 +1,41 @@
...
+import { useVaiVaultContract } from 'clients/contracts/hooks';
+
+type Options = MutationObserverOptions<
+  ClaimVaiVaultRewardOutput,
+  Error,
+  Omit<IClaimVaiVaultRewardInput, 'vaiVaultContract'>
+>;
+
+const useClaimVaiVaultReward = (options?: Options) => {
+  const vaiVaultContract = useVaiVaultContract();
```
Comment (coreyar) [inline src/clients/api/mutations/useClaimVaiVaultReward.ts:19]: Should we be using the ProxyContract here? 

We are checking for failure events but this contract doesn't send failure events. The Proxy contract does

Comment (therealemjy) [reply src/clients/api/mutations/useClaimVaiVaultReward.ts:19]: I don't think so, the current implementation doesn't use the proxy.
I can remove the checks in that case

Comment (coreyar) [reply src/clients/api/mutations/useClaimVaiVaultReward.ts:19]: In that case we don't have to check for the Failure event. The VAiVaultContract doesn't send the `fail` or `failOpaque` events. It doesn't hurt anything though

Comment (therealemjy) [reply src/clients/api/mutations/useClaimVaiVaultReward.ts:19]: Sounds good, I've removed the check

File: src/errors/transactionErrors.ts [hunk 2]
```diff
@@ -47,6 +55,12 @@ export const checkForTokenTransactionError = (receipt: TransactionReceipt) =>
...
-    VAIControllerErrorReporterError,
-    VAIControllerErrorReporterFailureInfo,
+    VaiControllerErrorReporterError,
+    VaiControllerErrorReporterFailureInfo,
   );
+
+export const checkForVaiVaultTransactionError = (receipt: TransactionReceipt) =>
+  checkForTransactionError(receipt, VaiVaultErrorReporterError, VaiVaultErrorReporterInfo);
+
+export const checkForXvsVaultTransactionError = (receipt: TransactionReceipt) =>
```
Comment (coreyar) [inline src/errors/transactionErrors.ts]: This is actually for checking the vaultProxyContract because the Vault Contract doesn't send these events

**Review by coreyar — APPROVED** [review]

**Pull request merged**